geode-core/src/main/java/org/apache/geode/management/cli/GfshCommand.java

```diff
@@ -250,4 +256,36 @@ public boolean poll(long timeout, TimeUnit unit, Supplier<Boolean> function) {
     return false;
 
   }
+
+  public boolean diskStoreBeanAndMemberBeanDiskStoreExists(DistributedSystemMXBean dsMXBean,
```

Comment:
`GfshCommand` represents all gfsh commands and is a User API for defining new User provided gfsh commands. Do you think it makes sense to expand the responsibilities of `GfshCommand` to include knowledge of DiskStore and JMX?

Since `GfshCommand` is not hidden behind an `internal` package, it is part of the User API. Exposing new public methods means we're exposing this to Users and providing guarantees of support and backwards compatibility.

You should probably look for an `internal` class that involves DiskStores and is within the context of JMX mbeans. For example, `org.apache.geode.management.internal` is the internal package containing classes (and sub-packages) that involve JMX.

Reply:
@kirklund  The code to check for DiskStores (in slightly varying ways) was previously located in a couple of subclasses of `GfshCommand` although the specific check methods were not publicly accessible. Since we wanted a unified way of checking the natural thing was to push them up the inheritance chain. 

After some digging I found `org.apache.geode.management.internal.cli.commands.DiskStoreCommandUtils` so that seems to be a good alternative to move the functionality of checking for DiskStores so I'll refactor to move the checks there.

Reply:
@kirklund pushed the updated code